packages/reactive-element/src/reactive-element.ts

```diff
@@ -737,6 +763,31 @@ export abstract class ReactiveElement
     return this.elementProperties.get(name) ?? defaultPropertyDeclaration;
   }
 
+  // This is a finalization step, but it needs to be separate from `finalize`
+  // because `finalize` can be called before a class has been fully initialized.
+  // This method should only be called once per class, as late as possible.
+  private static __collectMetadata() {
+    const metadata = this[Symbol.metadata];
+    if (metadata == null) {
+      return;
+    }
+    const properties = litPropertyMetadata.get(metadata);
+    if (properties === undefined) {
+      return;
+    }
+    for (const [p, options] of properties) {
+      this.elementProperties.set(p, options);
+      // If this class doesn't have its own set, create one and initialize
+      // with the values in the set from the nearest ancestor class, if any.
+      if (!this.hasOwnProperty('__reactivePropertyKeys')) {
```

Comment:
I believe this is exclusively for a DEV_MODE warning so should be gated for that setting.

Reply:
done